bot/cogs/codeblock/instructions.py

```diff
@@ -0,0 +1,177 @@
+            "helps improve the legibility and makes it easier for us to help you.\n\n"
+            f"**To do this, use the following method:**\n{example_blocks}"
+        )
+    else:
+        log.trace("Aborting missing code block instructions: content is not Python code.")
+
+
+def _get_bad_lang_message(content: str) -> Optional[str]:
+    """
+    Return instructions on fixing the Python language specifier for a code block.
+
+    If `code_block` does not have a Python language specifier, return None.
+    If there's nothing wrong with the language specifier, return None.
+    """
+    log.trace("Creating instructions for a poorly specified language.")
+
+    info = parsing.parse_bad_language(content)
+    if not info:
+        log.trace("Aborting bad language instructions: language specified isn't Python.")
+        return
+
+    lines = []
+    language = info.language
+
+    if info.leading_spaces:
+        log.trace("Language specifier was preceded by a space.")
+        lines.append(f"Make sure there are no spaces between the back ticks and `{language}`.")
+
+    if not info.terminal_newline:
+        log.trace("Language specifier was not followed by a newline.")
+        lines.append(
+            f"Make sure you put your code on a new line following `{language}`. "
+            f"There must not be any spaces after `{language}`."
+        )
+
+    if lines:
+        lines = " ".join(lines)
+        example_blocks = _get_example(language)
+
+        # Note that _get_bad_ticks_message expects the first line to have two newlines.
+        return (
+            f"It looks like you incorrectly specified a language for your code block.\n\n{lines}"
+            f"\n\n**Here is an example of how it should look:**\n{example_blocks}"
+        )
+    else:
+        log.trace("Nothing wrong with the language specifier; no instructions to return.")
+
+
+def _get_no_lang_message(content: str) -> Optional[str]:
+    """
+    Return instructions on specifying a language for a code block.
+
+    If `content` is not valid Python or Python REPL code, return None.
+    """
+    log.trace("Creating instructions for a missing language.")
+
+    if parsing.is_repl_code(content) or parsing.is_python_code(content):
+        example_blocks = _get_example("python")
+
+        # Note that _get_bad_ticks_message expects the first line to have two newlines.
+        return (
+            "It looks like you pasted Python code without syntax highlighting.\n\n"
+            "Please use syntax highlighting to improve the legibility of your code and make "
+            "it easier for us to help you.\n\n"
+            f"**To do this, use the following method:**\n{example_blocks}"
+        )
+    else:
+        log.trace("Aborting missing language instructions: content is not Python code.")
+
+
+def get_instructions(content: str) -> Optional[str]:
+    """Return code block formatting instructions for `content` or None if nothing's wrong."""
```

Comment:
I believe the docstring here could be a bit expanded, at first I found it a bit confusing that getting the instructions did the content checks on the text.

Reply:
Fixed in 621043a7ebc7574455394959a690913064100101